samples/queue/main.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * This is an example program which will go through using the FixedQueue
+ * type. The program will print out the results of the operations over UART
+ * for the user to watch the result.
+ *
+ * @author Collin Bolles
+ */
+#include <EVT/io/manager.hpp>
+#include <EVT/io/pin.hpp>
+#include <EVT/io/UART.hpp>
+#include <EVT/utils/types/FixedQueue.hpp>
+
+namespace IO = EVT::core::IO;
+namespace types = EVT::core::types;
+
+/**
+ * Test class for showing off having a custom class in the FixedQueue.
+ */
+class TestClass {
+ private:
+    int a;
+    int b;
+    int c;
+
+ public:
+    TestClass(int a, int b, int c) : a(a), b(b), c(c) {}
+    TestClass(): a(0), b(0), c(0) {}
+
+    // Copy operator, important to have
+    TestClass& operator=(const TestClass& other) {
+        if (this == &other)
+            return *this;
+        this->a = other.a;
+        this->b = other.b;
+        this->c = other.c;
+        return *this;
+    }
+
+    // For easier debugging
+    void print(IO::UART* uart) {
+        uart->printf("TestClass(a = %d, b = %d, c = %d)", a, b, c);
+    }
+};
+
+int main() {
+    // Setup UART
+    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Test a queue of numbers, no overwritting
+    ///////////////////////////////////////////////////////////////////////////
+    uart.printf("Queue of Numbers, no overwritting\n\r");
+    // First a queue of numbers, max capacity of 10 and no over writting
+    types::FixedQueue<int> numberQueue(10);
+    // Check if it is empty, should be empty
+    uart.printf("numberQueue.isEmpty() -> %d\r\n", numberQueue.isEmpty());
+
+    // Add ten numbers, the max allowed
+    for (int i = 0; i < 10; i++) {
```

Comment:
A little pedantic, but you don't try to over-fill the queue in the non-overwrite case, or pop from an empty queue. Just to be complete.

Reply:
I don't think that's too pedantic, the queue should be a known working thing. I'll add test cases for those